routes/location.js

```diff
@@ -35,24 +37,29 @@ function getCloudMetadata(callback) {
                     // Try Azure next
                     getAzureCloudMetadata(function(err, c, z) {
                         if (err) {
-                            // Try GCP next
                             getGCPCloudMetadata(function(err, c, z) {
-                                // Return result regardless of error
-                                callback(c, z); // Running in GCP or unknown
+                                if (err) {
+                                    // Try GCP next
+                                    getGeneric(function(err, c, z) {
```

Comment:
Consider naming this something specific to kubernetes e.g. `getKubeCloudMetadata`, `getK8sCloudMetadata`, etc.